src/platform/modulesmanager/src/ManagementModule.cpp

```diff
@@ -7,514 +7,476 @@
-const std::string GETMMIINFO_VERSIONINFO = "VersionInfo";
-const std::string GETMMIINFO_COMPONENTS = "Components";
-const std::string GETMMIINFO_LIFETIME = "Lifetime";
-const std::string GETMMIINFO_LICENSEURI = "LicenseUri";
-const std::string GETMMIINFO_PROJECTURI = "ProjectUri";
-const std::string GETMMIINFO_USERACCOUNT = "UserAccount";
+static const std::string g_mmiFuncMmiGetInfo = "MmiGetInfo";
+static const std::string g_mmiFuncMmiOpen = "MmiOpen";
+static const std::string g_mmiFuncMmiClose = "MmiClose";
+static const std::string g_mmiFuncMmiSet = "MmiSet";
+static const std::string g_mmiFuncMmiGet = "MmiGet";
+static const std::string g_mmiFuncMmiFree = "MmiFree";
+
+static const std::string g_mmiGetInfoName = "Name";
+static const std::string g_mmiGetInfoDescription = "Description";
+static const std::string g_mmiGetInfoManufacturer = "Manufacturer";
+static const std::string g_mmiGetInfoVersionMajor = "VersionMajor";
+static const std::string g_mmiGetInfoVersionMinor = "VersionMinor";
+static const std::string g_mmiGetInfoVersionPatch = "VersionPatch";
+static const std::string g_mmiGetInfoVersionTweak = "VersionTweak";
+static const std::string g_mmiGetInfoVersionInfo = "VersionInfo";
+static const std::string g_mmiGetInfoComponents = "Components";
+static const std::string g_mmiGetInfoLifetime = "Lifetime";
+static const std::string g_mmiGetInfoLicenseUri = "LicenseUri";
+static const std::string g_mmiGetInfoProjectUri = "ProjectUri";
+static const std::string g_mmiGetInfoUserAccount = "UserAccount";
 
 typedef void (*mmi_t)();
 
-ManagementModule::ManagementModule(const std::string clientName, const std::string path, const unsigned int maxPayloadSize) :
-    handle(nullptr),
-    mmiHandle(nullptr),
-    isValid(true),
-    clientName(clientName),
-    modulePath(path),
-    maxPayloadSizeBytes(maxPayloadSize)
+ManagementModule::ManagementModule() :
+    m_modulePath(""),
+    m_isValid(false),
+    m_handle(nullptr) {}
+
+ManagementModule::ManagementModule(const std::string path) :
+    m_modulePath(path),
+    m_isValid(true),
+    m_handle(nullptr)
 {
-    info.lifetime = Lifetime::Undefined;
-    info.userAccount= 0;
-    void* handle = dlopen(path.c_str(), RTLD_LAZY);
-    if (!ManagementModule::IsExportingMmi(path))
-    {
-        OsConfigLogError(ModulesManagerLog::Get(), "%s does not export the required Managament Module Interface (MMI)", path.c_str());
-        isValid = false;
-        return;
-    }
+    m_info.lifetime = Lifetime::Undefined;
+    m_info.userAccount= 0;
 
-    // Get the Management Module info and validate schema
-    Mmi_GetInfo mmiGetInfo = reinterpret_cast<Mmi_GetInfo>(dlsym(handle, MmiFuncMmiGetInfo.c_str()));
-    Mmi_Free mmiFree = reinterpret_cast<Mmi_Free>(dlsym(handle, MmiFuncMmiFree.c_str()));
-    MMI_JSON_STRING retJson = nullptr;
-    int retJsonSize = 0;
-    int retMmiGetInfo = MMI_OK;
-    ScopeGuard sg{[&]()
+    void* m_handle = dlopen(path.c_str(), RTLD_LAZY);
```

Comment:
> void* m_handle = dlopen(path.c_str(), RTLD_LAZY); [](http://example.com/codeflow?start=3&length=50)

Same comment.... Do not make calls that can fail from inside a constructor/destructor.